### PR TITLE
KT-22298 Improve docs for Array.copyOf(newSize: Int)

### PR DESCRIPTION
**Pull request by neonailol**
Issue: https://youtrack.jetbrains.com/issue/KT-22298 
Finished work started in #1477

### CODE REVIEW

File: libraries/tools/kotlin-stdlib-gen/src/templates/Arrays.kt [hunk 1]
```diff
@@ -443,7 +444,17 @@ object ArrayOps : TemplateGroupBase() {
...
+            """
+            Returns new array which is a copy of the original array, resized to the given [newSize], truncating or padding
+            with nulls if necessary. If the object is a primitive type, it will be padded with its default value.
+            For all indices that are valid in both the original array and the copy, the two arrays contents will be identical.
+            For all indices that are valid in the copy, but not in the original, the copy will contain either null or the default
+            value.
+            """
+        }
+        sample("samples.collections.Arrays.CopyOfOperations.resizingCopyOf")
+        sample("samples.collections.Arrays.CopyOfOperations.resizedPrimitiveCopyOf")
```
Comment (ilya-g) [inline libraries/tools/kotlin-stdlib-gen/src/templates/Arrays.kt]: Better to provide each sample only to the related overload of `copyOf`

File: libraries/tools/kotlin-stdlib-gen/src/templates/Arrays.kt [hunk 2]
```diff
@@ -443,7 +444,17 @@ object ArrayOps : TemplateGroupBase() {
         include(ArraysOfPrimitives, PrimitiveType.defaultPrimitives)
         include(InvariantArraysOfObjects)
     } builder {
-        doc { "Returns new array which is a copy of the original array, resized to the given [newSize]." }
+        doc {
+            """
+            Returns new array which is a copy of the original array, resized to the given [newSize], truncating or padding
+            with nulls if necessary. If the object is a primitive type, it will be padded with its default value.
```
Comment (ilya-g) [inline libraries/tools/kotlin-stdlib-gen/src/templates/Arrays.kt]: Please add a blank line to separate summary and description.
The text can be specialized for primitives based on the current value of overloads `family`.

File: libraries/stdlib/samples/test/samples/collections/arrays.kt [hunk 3]
```diff
@@ -76,4 +76,28 @@ class Arrays {
...
+            val array = arrayOf("apples, oranges, lime")
+            val arrayCopy = array.copyOf()
+            assertPrints(arrayCopy.contentToString(), "[apples, oranges, lime]")
+        }
+
+        @Sample
+        fun resizingCopyOf() {
+            val array = arrayOf("apples, oranges, lime")
+            val arrayCopyResized = array.copyOf(5)
+            assertPrints(arrayCopyResized.contentToString(), "[apples, oranges, lime, null, null]")
```
Comment (ilya-g) [inline libraries/stdlib/samples/test/samples/collections/arrays.kt]: Worth to add truncating sample too.

File: libraries/tools/kotlin-stdlib-gen/src/templates/Arrays.kt (hunk 2 above)
Comment (neonailol) [reply libraries/tools/kotlin-stdlib-gen/src/templates/Arrays.kt]: About text specialization, I am not sure how to approach this without coping description, the only thing I found is `textWhen` method, but with it description looks messy.

```
            """
            Returns new array which is a copy of the original array, resized to the given [newSize],
            truncating or padding with ${textWhen(family == ArraysOfPrimitives) { "primitive default value" }}${textWhen(family == InvariantArraysOfObjects) { "null" }} if necessary.

            For all indices that are valid in both the original array and the copy, the two arrays contents will be identical.
            For all indices that are valid in the copy, but not in the original, the copy will contain ${textWhen(family == ArraysOfPrimitives) { "primitive default value" }}${textWhen(family == InvariantArraysOfObjects) { "null" }}.
            """
```

Comment (ilya-g) [reply libraries/tools/kotlin-stdlib-gen/src/templates/Arrays.kt]: If `textWhen` is messy, you can use plain `if` or `when` expression inside `${...}`

**Comment by neonailol** [comment]
@ilya-g Implemented all requested changes

**Comment by ilya-g** [comment]
Thanks, I've merged new docs with some edits.

**Pull request closed without merging**